### ISSUE TITLE
Which epoch of text encoder and image encoder to use ?

### ISSUE THREAD

**Issue by ghost** [sidward14/Style-AttnGAN#29]
Hello! I pre-trained the DAMSM + GPT-2 model for 600 epochs, and I have saved files for the text encoder and image encoder from `text_encoder50.pth` and `image_encoder50.pth` to `image_encoder550.pth` and `image_encoder550.pth`. I want to ask which epoch I should use for both the text encoder and image encoder to train the GAN

**Comment by sidward14** [comment]
I would look at the generated images with your own eyes to gauge which epoch to train some more from. That's what I ended up doing in the end.

It's hard to quickly gauge a GAN's performance from quantitative measures. Even metrics like FID and PPL are unsatisfactory a lot of the time. I've seen many occasions where they are decreasing, yet if you look at the generated images, it tells a different story.

But if you don't have time for that: generally, the latest saved models are the best ones unless there's something like modal collapse (where it seems like all the text inputs are leading to the same generated image). I don't remember if 550 epochs was too much for modal collapse; I don't think it was.

**Comment by ghost** [comment]
I think this guy @VeryIdiotUser is trying to ask which epoch for the text encoder DAMSM to use in order to train the GAN model, he hasn't trained the GAN yet, the process is to train the DAMSM first then pick the best epoch of the DAMSM to train the GAN, 

I also don't know which epoch for the DAMSM to use after training for 600 epochs, there are outputs like the one from @VeryIdiotUser: `valid loss`, `s_loss`, `w_loss`, should I choose based on these statistics, or based on something else ?

**Comment by sidward14** [comment]
Ohh didn't read that part. In that case, use the valid loss. The first number is the valid `s_loss` while the second number is the valid `w_loss`.

In my experience, using transformer for comparing text and image embeddings doesn't work as well as you'd expect. The BiLSTM worked better. OpenAI found similar results: https://arxiv.org/pdf/2103.00020.pdf  (Fig 2)
